valkyrie/lib/valkyrie/storage/disk.rb

```diff
@@ -9,10 +9,11 @@ def initialize(base_path:, path_generator: BucketedStorage, file_mover: FileUtil
     end
 
     # @param file [IO]
+    # @param original_filename [String]
     # @param resource [Valkyrie::Resource]
     # @return [Valkyrie::StorageAdapter::File]
-    def upload(file:, resource: nil)
-      new_path = path_generator.generate(resource: resource, file: file)
+    def upload(file:, original_filename:, resource: nil)
```

Comment:
Should we make the default value for `original_filename` be `file.original_filename`?  It seems like having a default would be nice to avoid issues in `generate` if it was nil.

Reply:
I don't think so because this implies that `file` will be an `ActionDispatch::HTTP::UploadedFile` which is not always the case.  I suspect you will always use `FileAppender`, so you wouldn't ever have to worry as all the changes have been encapsulated.

Reply:
OK, so in practice this won't be an issue.  But I do think it's important to prevent a nil original_filename because it will make uploads fail (or worse, succeed, but have subsequent uploads with an original_filename specified fail).

So maybe `BucketedStorage#generate` should raise an error if `original_filename` is nil?

Reply:
Good call. I've added an ArgumentException to that method.